### PR TITLE
code refactored; addjob button function added

### PR DESCRIPTION
**Pull request by EgemenUslu**

### CODE REVIEW

File: src/components/Card/Card.jsx [hunk 1]
```diff
@@ -6,20 +6,20 @@ const Card = (props) => {
   const {card: {companyLogo, company, jobTitle, isRejected}} = props;
 
   return (
-    <div className={isRejected ? "card-rejected-container" : "card-container" }>
+    <div className="card-container" style={isRejected ? {backgroundColor: "rgba(188,43,255,.4)", textAlign: "left" } : {} }>
```
Comment (winterchagas) [inline src/components/Card/Card.jsx]: We can make dynamic classes instead of forcing css on js.

File: src/components/Column/Column.jsx [hunk 2]
```diff
@@ -30,50 +40,64 @@ class Column extends Component {
...
-            {this.props.title}
-            ({this.props.totalCount})
-          </div>
-          <div className="column-header column-details">
-            {this.props.details}
-          </div>          
-        </div>
-        <div className="column-addjob">
-          {this.addJob()}
+    <div className="column-header-container"  style= {this.state.isAddJobClicked ? {height:180}:{}} >
```
Comment (winterchagas) [inline src/components/Column/Column.jsx]: Same as previous comment

File: src/components/Column/style.scss [hunk 3]
```diff
@@ -1,165 +1,184 @@
...
+    border-top-left-radius: 4px;
+    border-top-right-radius: 4px;
+    padding-bottom: 15px;
 
+    &.no-card {
+      border-radius: 5px;  
+    }
+    
     .column-header {
+      font-family: Exo;
```
Comment (winterchagas) [inline src/components/Column/style.scss]: Are we going to have different fonts in the project?
Let's set in the body. So we don't have to set it any more times.

File: src/components/Column/style.scss [hunk 4]
```diff
@@ -1,165 +1,184 @@
...
+      margin-left: 15px;
+      margin-top: 30px;
+      width: 240px;
+      height: 70px;
+      border: 1px solid rgb(255, 255, 255);
+      border-radius: 4px;
+      box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.1);
+      background-color: rgb(255, 255, 255);
+      font-family: Exo;
+      font-style: normal;
```
Comment (winterchagas) [inline src/components/Column/style.scss]: I think all these properties are default values. No need.

File: src/components/Column/style.scss [hunk 5]
```diff
@@ -1,165 +1,184 @@
...
       border: none;
-      margin-left: 43%;
-      background-color: rgba(255, 255, 255, 1);
-      color: rgba(150, 150, 150, 0.911);
-      font-family: sans-serif;
-      font-weight: bold;
-      font-size: 90%;
+      background-color: rgb(228, 220, 255);  
+      font-size: 18px;
+      font-weight: normal;
```
Comment (winterchagas) [inline src/components/Column/style.scss]: Same thing.

File: src/components/DetailsModal/DetailsModal.jsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    return (
+      <main>
+        {this.Modal()}
+        <div className="show-modal-button" onClick={this.toggleModal}>x</div>
+      </main>
+    )
+  }
+}
+
+export  { DetailsModal , toggleModal};
```
Comment (winterchagas) [inline src/components/DetailsModal/DetailsModal.jsx]: This is looking like an anti pattern.

File: src/components/Card/Card.jsx [hunk 7]
```diff
@@ -1,26 +1,28 @@
...
+import { DetailsModal , toggleModal} from '../DetailsModal/DetailsModal.jsx';
 
 import './style.scss'
 
 const Card = (props) => {
   const {card: {companyLogo, company, jobTitle, isRejected}} = props;
 
   return (
-    <div className="card-container" style={isRejected ? {backgroundColor: "rgba(188,43,255,.4)", textAlign: "left" } : {} }>
+    <div className={isRejected ? "card-container rejected-cards" : "card-container"} >
```
Comment (winterchagas) [inline src/components/Card/Card.jsx]: className=`card-container ${isRejected ? 'rejected-cards' : ''}`

Comment (winterchagas) [reply src/components/Card/Card.jsx]: Same thing on the other places we have this pattern.

Comment (winterchagas) [reply src/components/Card/Card.jsx]: There are backticks after = and at the end, github deletes the character

File: src/components/DetailsModal/DetailsModal.jsx (hunk 6 above)
Comment (winterchagas) [reply src/components/DetailsModal/DetailsModal.jsx]: Where are we using the toggle?

File: src/components/Column/Column.jsx [hunk 8]
```diff
@@ -64,7 +64,7 @@ class Column extends Component {
 
   generateColumnHeader() {
     return (
-    <div className="column-header-container"  style= {this.state.isAddJobClicked ? {height:180}:{}} >
+    <div className={this.props.totalCount == MIN_CARD_NUMBER_IN_COLUMN ? "column-header-container no-card" : "column-header-container"}  style= {this.state.isAddJobClicked ? {height:180}:{}} >
```
Comment (winterchagas) [inline src/components/Column/Column.jsx]: Are they the same type? If so, use  ===

File: src/components/DetailsModal/DetailsModal.jsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    this.toggleModal = this.toggleModal.bind(this);
+  }
+
+  toggleModal () {
+    this.setState(state => ({
+      isModalShowing: !state.isModalShowing
+    }));
+  }
+
+  Modal() {
```
Comment (winterchagas) [inline src/components/DetailsModal/DetailsModal.jsx]: Why is it in a separate function?

File: src/components/Card/Card.jsx (hunk 7 above)
Comment (EgemenUslu) [reply src/components/Card/Card.jsx]: I need help for this one, it did not work or I could not do it in right way.

File: src/components/Column/Column.jsx (hunk 2 above)
Comment (EgemenUslu) [reply src/components/Column/Column.jsx]: Might you show me how to create the dynamic css function in .jsx again?

File: src/components/DetailsModal/DetailsModal.jsx (hunk 6 above)
Comment (EgemenUslu) [reply src/components/DetailsModal/DetailsModal.jsx]: Modal structure is gonna change I just pushed as it was yesterday. I need to research it more.

File: src/components/DetailsModal/DetailsModal.jsx (hunk 9 above)
Comment (EgemenUslu) [reply src/components/DetailsModal/DetailsModal.jsx]: Modal structure is gonna change I just pushed as it was yesterday. I need to research it more.

**Pull request merged**